src/main/java/joe/commands/MarkCommand.java

```diff
@@ -0,0 +1,42 @@
+package joe.commands;
+
+import joe.Storage;
+import joe.TaskList;
+import joe.Ui;
+import joe.exceptions.JoeIndexOutOfBoundsException;
+
+/**
+ * Represents a command to mark a task as done in the task list.
+ */
+public class MarkCommand extends Command {
+    private final int idx;
```

Comment:
Perhaps we could use a more intuitive variable name here? Something like `taskIndex`.